airbyte-ci/connectors/pipelines/pipelines/helpers/utils.py

```diff
@@ -101,13 +101,8 @@ async def get_file_contents(container: Container, path: str) -> Optional[str]:
     Returns:
         Optional[str]: The file content if the file exists in the container, None otherwise.
     """
-    try:
-        return await container.file(path).contents()
-    except QueryError as e:
-        if "no such file or directory" not in str(e):
-            # this error could come from a network issue
-            raise
-    return None
+    nonexistence = await container.with_exec(["sh", "-c", f"[ -f '{path}' ] || echo '{path} does not exist'"]).stdout()
```

Comment:
This consider that:
* `sh` is available in the container
* the container entrypoint is not overidden (you could set `skip_entrypoint=True` on the`with_exec`)

I'd find it more robust to only rely on Dagger's API to check file existence.
Maybe something like:
* Inferring the directory from the path
* Call `await container.directory(directory_path).entries()` and check if the file is in the returned list.

Reply:
Yeah, I didn't think this one through. Good catch!